### PR TITLE
Update to 1.18.2

### LINKED ISSUE #58: 1.18.2 not marked as supported
i wanted to add this to my modpack but i coudnt because it is only marked for up to 1.18.1, not 1.18.2 so i cant add it to my pack unfortunately. :(

### PR DESCRIPTION
**Pull request by ThexXTURBOXx**
Initial update to 1.18.2.
However, there is still one TODO missing which I cannot fix myself.

**Download:** https://github.com/theorbtwo/RoughlyEnoughResources/suites/5867622146/artifacts/197914301

Fixes #58

### CODE REVIEW

File: src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java [hunk 1]
```diff
@@ -228,7 +225,7 @@ public void munchLootCondition(JsonElement conditionElement, List<LootOutput> ou
                     outputs.addAll(munchLootSupplierJson(json));
                 break;
             case "minecraft:tag":
-                Tag<Item> tag = ItemTags.getTagGroup().getTag(new Identifier(object.get("name").getAsString()));
+                Tag<Item> tag = null;// TODO ItemTags.getTagGroup().getTag(new Identifier(object.get("name").getAsString()));
```
Comment (ThexXTURBOXx) [inline src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java]: I am unsure how to fix this. ``ItemTags#getTagGroup`` has been removed in 1.18.2 and I cannot seem to find a replacement

Comment (North-West-Wind) [reply src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java]: Funny enough, [an article](https://gist.github.com/SizableShrimp/ddcbe9a9862cc4a0f526c42ae49b2c1d) on the Forge side seems to be helpful.

Anyway, I think you may want to look into the `Registry` class. It has a `streamTags` method but I don't know if that's what you are finding.

**Comment by JimmyTheGreat1500** [comment]
So does it work on 1.18.2? reading through mojang's change logs and i dont see anything that i imagine will break it... idk?

**Comment by ThexXTURBOXx** [comment]
@JimmyTheGreat1500 As stated above, the whole `Tags` system got reworked. So, there is quite a lot that is broken :)

File: src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java (hunk 1 above)
Comment (ThexXTURBOXx) [reply src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java]: Thanks for the article! However, I still don't know exactly how to fix this, sadly. Maybe I will read through the internals to see what I could do instead

Comment (ThexXTURBOXx) [reply src/main/java/uk/me/desert_island/rer/rei_stuff/LootDisplay.java]: Okay, I got it!
The `streamTagsAndEntries` method did the trick. Together with some converting and mapping streams, it is finally working again!
Now, this PR is complete, thank you very much! :)

**Comment by ThexXTURBOXx** [comment]
@theorbtwo This PR is now complete and ready to get merged!

**Comment by JimmyTheGreat1500** [comment]
Ohh, ok. Now I see. When do you estimate it will be released?

**Comment by ThexXTURBOXx** [comment]
I am not the creator of this mod, I am just some guy that wanted to update this mod :)
Most likely no one except for the creators will be able to give you an ETA

**Comment by shedaniel** [comment]
Thank you for your contributions!

**Pull request merged**